src/query/phrase_query/phrase_scorer.rs

```diff
@@ -236,13 +190,22 @@ impl<TPostings: Postings> PhraseScorer<TPostings> {
     }
 
     fn phrase_exists(&mut self) -> bool {
+        // there's a possible optim:
+        // compute_phrase_match until the penultimate term
+        // re implement the matching:
+        //   * match self.left with the ultimate term
+        //   * exit fast on first intersection
         let intersection_len = self.compute_phrase_match();
-        intersection_exists(&self.left[..intersection_len], &self.right[..])
+        intersection_len > 0
     }
 
     fn compute_phrase_count(&mut self) -> u32 {
-        let intersection_len = self.compute_phrase_match();
-        intersection_count(&self.left[..intersection_len], &self.right[..]) as u32
+        // there's a possible optim:
+        // compute_phrase_match until the penultimate term
+        // re implement the matching:
+        //   * match self.left with the ultimate term
+        //   * count without storing in left
+        self.compute_phrase_match() as u32
```

Comment:
I can add the implems for both optims here, just wanted to know if this looked better now and if this was necessary or not :)